sklearn/metrics/pairwise.py

```diff
@@ -935,13 +939,46 @@ def _parallel_pairwise(X, Y, func, n_jobs, **kwds):
     if Y is None:
         Y = X
 
+    if n_jobs == 1:
```

Comment:
It would be worth adding an inline comment to explain that as it's far from trivial when reading the source.